### PR TITLE
Add README on basics for how to read Orca memos

### PR DESCRIPTION
**Pull request by dgkimura**
While learning basics how to read Orca memos, I did not find any readily
available documentation. This is an attempt to leave something behind for the
next person.

https://github.com/dgkimura/gpdb/blob/memo-readme/src/backend/gporca/README.memo.md

### CODE REVIEW

File: src/backend/gporca/README.memo.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
+# Understanding Orca Memos
+
+Orca memos is a structure that enumerates different, but logically identical
```
Comment (ashuka24) [inline src/backend/gporca/README.memo.md:3]:
``` suggestion
An Orca memo is a structure that enumerates different, but logically identical
```

File: src/backend/gporca/README.memo.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+plans and then optimizes to find the most efficient one. Let's explore how to
```
Comment (ashuka24) [inline src/backend/gporca/README.memo.md:4]:
```suggestion
plans, and then optimizes to find the most efficient one. Let's explore how to
```

File: src/backend/gporca/README.memo.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+Apply following GUCs in psql session:
+```sql
+-- These GUCS print the memo structure to the console and provide the necessary
+-- information to follow the logic to derive the optimal plan
+SET client_min_messages=log;
+SET optimizer_print_memo_after_optimization=on;
+SET optimizer_print_optimization_context=on;
+```
+
```
Comment (ashuka24) [inline src/backend/gporca/README.memo.md:17]: Maybe you can add a comment that you can also read memo after exploration and memo after implementation

File: src/backend/gporca/README.memo.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+the memo:
+```
+CPhysicalMotionGather(master)
++--CPhysicalInnerHashJoin
+   |--CPhysicalTableScan "t" ("t")
+   |--CPhysicalTableScan "t" ("t")
+   +--CScalarCmp (=)
+      |--CScalarIdent "a" (0)
+      +--CScalarIdent "a" (8)
+```
```
Comment (ashuka24) [inline src/backend/gporca/README.memo.md:255]: Maybe add a comment on what it means if there is no best expression to be found and what to do in that case,  no cost context to be found, or what it means when it says `PRUNED`

**Review by ashuka24 — COMMENTED** [review]
Thank you so much for writing this up and putting in so much detail!

**Review by zellerh — APPROVED** [review]
Thanks for adding this very helpful info!

File: src/backend/gporca/README.memo.md (hunk 4 above)
Comment (dgkimura) [reply src/backend/gporca/README.memo.md:263]: I couldn't think of a good place to add this without going on a tangent.  I'm going to leave it for now and think of how to revisit later.

File: src/backend/gporca/README.memo.md (hunk 3 above)
Comment (dgkimura) [reply src/backend/gporca/README.memo.md:17]: Thanks for suggestion.  Done

**Comment by dgkimura** [comment]
Thanks for the reviews!

File: src/backend/gporca/README.memo.md (hunk 4 above)
Comment (ashuka24) [reply src/backend/gporca/README.memo.md:263]: Sounds good. Thanks so much for writing this up!

**Pull request merged**